### PR TITLE
daemon: make activation optional

### PR DESCRIPTION
**Pull request by chipaca**
offered as-is. Could also add spread tests, after the standup, or spread tests can be added in a later branch.

### CODE REVIEW

File: daemon/daemon.go [hunk 1]
```diff
@@ -175,6 +176,42 @@ func logit(handler http.Handler) http.Handler {
...
+	if err != nil {
+		return nil, err
+	}
+
+	listener, err := net.ListenUnix("unix", address)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := os.Chmod(socketPath, 0666); err != nil {
```
Comment (pedronis) [inline daemon/daemon.go]: @chipaca wondering, is this a bit racy? is there a best/better practice?

Comment (chipaca) [inline daemon/daemon.go]: it is. And I was about to say no, but yes. Pushing.

File: daemon/daemon.go [hunk 2]
```diff
@@ -175,6 +177,39 @@ func logit(handler http.Handler) http.Handler {
...
+	if err := os.Remove(socketPath); err != nil && !os.IsNotExist(err) {
+		return nil, err
+	}
+
+	address, err := net.ResolveUnixAddr("unix", socketPath)
+	if err != nil {
+		return nil, err
+	}
+
+	oldmask := unix.Umask(0111)
```
Comment (pedronis) [inline daemon/daemon.go]: I think you want runtime.LockOSThread and friend then

File: daemon/daemon.go [hunk 3]
```diff
@@ -184,24 +222,28 @@ func (d *Daemon) Init() error {
...
-	// the listeners, so check it before using it.
-	d.snapListener = listenerMap[dirs.SnapSocket]
+	if listener, err := getListener(dirs.SnapSocket, listenerMap); err == nil {
+		// Note that the SnapSocket listener does not use ucrednet. We use the lack
+		// of remote information as an indication that the request originated with
+		// this socket. This listener may also be nil if that socket wasn't among
+		// the listeners, so check it before using it.
+		d.snapListener = listener
+	} else {
+		logger.Debugf("cannot get listener for %q: %v", dirs.SnapSocket, err)
```
Comment (pedronis) [inline daemon/daemon.go]: should we turn this into  a real error now ?

Comment (chipaca) [inline daemon/daemon.go]: maybe leave that for after the cut?

Comment (pedronis) [inline daemon/daemon.go]: as you prefer, think is best

**Pull request merged**